### PR TITLE
Running dev without pro

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Exclude backend-core from the bundles if pro is not available locally.

If pro is available locally everything is bundled in a single file, so all the packages will be using the common bundled backend-core. If it's not available, pro will be using the backend-core from `backend-core/dist/`, while the rest will be using the bundled. Having both will cause issues with the main context:
image

Tests:
1. Running with the submodule available: ✅ https://github.com/Budibase/budibase/actions/runs/5667235977/job/15355559651
2. Running without the submodule: ✅ https://github.com/Budibase/budibase/actions/runs/5667345456/job/15355907172

### CODE REVIEW

File: packages/worker/package.json [hunk 1]
```diff
@@ -104,5 +104,19 @@
     "typescript": "4.7.3",
     "update-dotenv": "1.1.1"
   },
+  "nx": {
```
Comment (adrinr) [inline packages/worker/package.json:107]: If we don't have pro, backend-core needs to be built in order to run

**Review by mike12345567 — APPROVED** [review]
Nicely worked out! Makes sense to exclude from build if pro contains it.

File: package.json [hunk 2]
```diff
@@ -51,9 +51,9 @@
     "kill-builder": "kill-port 3000",
     "kill-server": "kill-port 4001 4002",
     "kill-all": "yarn run kill-builder && yarn run kill-server",
-    "dev": "yarn run kill-all && lerna run --stream dev:builder --stream",
-    "dev:noserver": "yarn run kill-builder && lerna run --stream dev:stack:up && lerna run --stream dev:builder --ignore @budibase/backend-core --ignore @budibase/server --ignore @budibase/worker",
-    "dev:server": "yarn run kill-server && lerna run --stream dev:builder --scope @budibase/worker --scope @budibase/server",
+    "dev": "yarn run kill-all && yarn nx run-many --target=dev:builder",
```
[the remark this answers is not in the document]
Comment (Rory-Powell) [reply package.json:54]: Does lerna not honour it? 🤯 
We should probably replace all `lerna run` in that case

[the remark this answers is not in the document]
Comment (adrinr) [reply package.json:54]: Yes, I felt the same 🤯

**Comment by adrinr** [comment]
Merging it to master as it's causing noise on the pipelines

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrinr) [reply package.json:49]: False alarm. After testing it with clean environments it worked as expected. I ran into the issue while loading/unloading the submodule, so something might be wrong in my environment. After some more checks, all work as expected. This changes had been reverted https://github.com/Budibase/budibase/pull/11380

**Pull request merged**